lib/src/client/impl/channel_impl.dart

```diff
@@ -579,6 +580,12 @@ class _ChannelImpl implements Channel {
       ..arguments = arguments;
 
     Completer<Exchange> opCompleter = Completer<Exchange>();
+
+    if (!declare) {
```

Comment:
Can you please add a test case similar to [this](https://github.com/achilleasa/dart_amqp/blob/master/test/lib/queue_test.dart#L105-L130) in the exchange test [suite](https://github.com/achilleasa/dart_amqp/blob/master/test/lib/exchange_test.dart)?

Reply:
Done :)